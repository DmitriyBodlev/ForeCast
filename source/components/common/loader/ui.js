import styled, { keyframes } from 'react-emotion';

export const LoaderWrapper = styled.div`
  align-items: center;
  background: rgba(53, 53, 53, 0.5);
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: center;
  position: fixed;
  top: 0;
  width: 100%;
`;

export const DimmerWrapper = styled.div`
  height: 100%;
  left: 0;
  top: 0;
  padding: 20px;
  position: absolute;
  width: 100%;
  z-index: 11;

  background-color: white;
`;

export const TitleRectangle = styled.div`
  height: 20px;
  margin: 20px 10px;
  width: 50px;

  background-color: grey;
`;

export const Rectangle = styled.div`
  height: 20px;
  margin-bottom: 20px;
  width: 100%;

  background-color: grey;
`;

export const TableRectangle = styled.div`
  margin-bottom: 20px;
  padding: 20px;
  width: 550px;

  background-color: lightgrey;
`;

export const squareRotate = keyframes`
  0% { 
    transform: perspective(120px) rotateX(0deg) rotateY(0deg);
  } 50% { 
    transform: perspective(120px) rotateX(-180.1deg) rotateY(0deg);
  } 100% { 
    transform: perspective(120px) rotateX(-180deg) rotateY(-179.9deg);
  }
`;

export const LoaderContent = styled.div`
  height: 40px;
  padding: 15px;
  border-radius: 2px;
  position: fixed;
  width: 40px;

  background-color: red;

  animation: ${squareRotate} 1.2s infinite ease-in-out;
  
`;
