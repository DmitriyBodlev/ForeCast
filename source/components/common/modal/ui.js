import styled, { css, keyframes } from 'react-emotion';

export const animation = (props: Object) => {
  const from = props.params.position.split('-')[0];

  const moves = {
    top: keyframes`
      0% { top: -${props.params.height}px }
      100% { top: 0 }
    `,
    bottom: keyframes`
      0% { bottom: -${props.params.height}px }
      100% { bottom: 0 }
    `,
  };

  return moves[from];
};

export const ModalWrapper = styled.div`
  position: fixed;
  width: 100%;
  height: 100%;
  top: 0;
  background: rgba(53, 53, 53, 0.5);
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
`;

export const Title = styled.span`
`;

export const ModalContent = styled.div`
  padding: 15px;
  position: fixed;
  border-radius: 2px;
  width: ${({params}: Object) => params.width}px;
  height: ${({params}: Object) => params.height}px;
  background-color: grey;
  box-shadow: 0 0 8px 1px black;
  animation: ${(props: Object) => animation(props)} 0.2s linear;
`;

export const Header = styled.div`
  font-weight: bold;
  color: black;
  font-size: 17px;
  align-items: center;
  display: inline-flex;
  width: 100%;
  justify-content: space-between;
`;

export const Footer = styled.div`
  position: absolute;
  bottom: 0px;
  width: 100%;
  border-top: 1px solid;
  height: 55px;
  padding: 10px 15px;
`;

const topLeft = css`
  top: 0;
  left: 0;
`;
const topCenter = css`
  top: 0;
  right: 50%;
  transform: translateX(50%);
`;
const topRight = css`
  top: 0;
  right: 0;
`;

const bottomLeft = css`
  bottom: 0;
  left: 0;
`;

const bottomCenter = css`
  bottom: 0;
  right: 50%;
  transform: translateX(50%);
`;

const bottomRight = css`
  bottom: 0;
  right: 0;
`;

const classStyles = {
  'top-left': topLeft,
  'top-center': topCenter,
  'top-right': topRight,
  'bottom-left': bottomLeft,
  'bottom-center': bottomCenter,
  'bottom-right': bottomRight,
};

export default classStyles;
