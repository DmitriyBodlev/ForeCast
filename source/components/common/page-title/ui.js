import { css } from 'react-emotion';

const wrapper = css`
  font-size: 20px;
  font-weight: 500;
  height: 60px;
  line-height: 60px;
  padding-left: 15px;

  background-color: white;
  color: black;
`;

const styles = {
  wrapper,
};

export default styles;
