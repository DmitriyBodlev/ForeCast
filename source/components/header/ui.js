import styled, { css } from 'react-emotion';

export const Divider = styled.span`
  display: inline-block;
  background-color: black;
  height: 32px;
  width: 1px;
  margin: auto 7px;
`;

export const HeaderWrapper = styled.div`
  background-color: grey;
  height: 100px;
  padding-right: 15px;
  width: 100%;
`;

export const Logo = styled.img`
  max-height: 70px;
  max-width: 100%;
  vertical-align: middle;
`;

const helperStyle = css`
  display: inline-block;
  height: 100%;
  vertical-align: middle;
`;

const logoStyle = css`
  float: left;
  margin-right: 3%;
  height: 70px;
  max-width: 15%;
  text-align: center;
  white-space: nowrap;
`;

const brandStyle = css`
  height: inherit;
  padding: 15px;
  position: relative;
`;

const textStyle = css`
  display: inline-block;
  height: 70px;
  line-height: 70px;
  text-align: left;
  vertical-align: middle;
  width: 39%;
`;

const menuStyle = css`
  display: flex;
  justify-content: flex-end;
  float: right;
  line-height: 15px;
  height: 30px;
  margin: 20px 0;
  width: 42%;
`;

const classStyle = {
  brand: brandStyle,
  helper: helperStyle,
  logo: logoStyle,
  menu: menuStyle,
  text: textStyle,
};

export default classStyle;
