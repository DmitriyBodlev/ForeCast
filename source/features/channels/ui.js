import styled from 'react-emotion';

export const BunnerWrapper = styled.div`
  background: #ffffff url(${(props: Object) => props.image}) no-repeat center center;
  background-size: cover;
  width: 100vw;
  height: 450px;
  display: flex;
  justify-content: center;
  align-items: center;
  & img {
    width: auto;
  }
`;

export const BlaBla = styled.div`
  height: 200px;
  width: 200px;
  background-color: black;
  color: white;
`;
