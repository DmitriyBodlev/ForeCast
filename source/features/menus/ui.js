import styled from 'react-emotion';
import { withTheme } from 'theming';

export const NavContainer = withTheme(styled.nav`
  box-shadow: 'inset 0 0 2px 1px rgba(217, 217, 217, 0.5)';
  display: flex;
  height: 40px;
  cursor: pointer;
  font-size: 12px;
  font-weight: 400;
  line-height: 40px;

  transition: margin-left .2s linear;

  margin-left: ${(props: Object) => props.openedSidebar ? '210px' : '0px'};
  background-color: ${({theme}: Object) => theme.menus.nav.bgColor};
  color: ${({theme}: Object) => theme.menus.nav.itemTextColor};
`);

const renderBackgroundIcon = (props: Object) => {
  if (props.currentTheme === 'dark') {
    return props.darkLink;
  }
  return props.lightLink;
};

export const NavItem = withTheme(styled.div`
  background-repeat: no-repeat;
  background-position-y: center;
  background-position-x: 15px;
  padding: 0 10px 0 45px;

  background-image: url(${(props: Object) => renderBackgroundIcon(props)});

  &:hover {
    background-color: ${({theme}: Object) => theme.menus.nav.bgColorHovered};
  }
`);

export const SidebarContainer = withTheme(styled.aside`
  box-shadow: 'inset 0 0 2px 1px rgba(217, 217, 217, 0.5)';
  cursor: default;
  font-size: 14px;
  height: calc(100vh - 87px);
  position: absolute;
  left: -210px;
  width: 210px;

  transition: left .2s linear;

  background-color: ${({theme}: Object) => theme.menus.sidebar.bgColor};
  color: ${({theme}: Object) => theme.menus.sidebar.itemTextColor};

  &.opened {
    left: 0px;
  }
`);

export const SidebarHeader = withTheme(styled.div`
  background-repeat: no-repeat;
  background-position-y: center;
  background-position-x: 180px;
  height: 40px;
  padding: 0 10px 0 45px;

  background-image: url('https://s3.amazonaws.com/amous-front-dev/icons/close.svg');

  &:hover {
    background-color: ${({theme}: Object) => theme.menus.sidebar.bgColorHovered};
  }
`);

const renderSidebarGroupWrapperBgColor = (props: Object) => {
  if (props.opened) {
    return props.theme.menus.sidebar.bgColorHovered;
  }
  return props.theme.menus.sidebar.bgColor;
};

export const SidebarGroupWrapper = withTheme(styled.div`
  background-color: ${(props: Object) => renderSidebarGroupWrapperBgColor(props)};
`);

export const SidebarGroupSubitemsWrapper = withTheme(styled.div`
`);

export const SidebarItem = withTheme(styled.div`
  background-repeat: no-repeat;
  background-position-y: center;
  background-position-x: 15px;
  cursor: pointer;
  height: 40px;
  line-height: 40px;
  padding: 0 10px 0 45px;

  background-image: url(${(props: Object) => renderBackgroundIcon(props)});
  font-weight: ${({opened}: Bool) => opened && 'bold'};
  color: ${(props: Object) => props.opened ? props.theme.menus.sidebar.itemTextColorHovered : props.theme.menus.sidebar.itemTextColor};

  &:hover {
    background-color: ${({theme}: Object) => theme.menus.sidebar.bgColorHovered};
    color: ${({theme}: Object) => theme.menus.sidebar.itemTextColorHovered};
  }
`);

export const SidebarItemRightPlusIcon = withTheme(styled.span`
  &::after { 
    content: "+";
    color: #ce2828;
    font-size: 20px;
    font-weight: 900;
    float: right;
  }
`);

const renderArrowIconRotate = ({opened}: boolean) => {
  if (opened) {
    return 'rotate(-45deg)';
  }
  return 'rotate(135deg)';
};

export const SidebarItemRightArrowIcon = withTheme(styled.span`
  &::after {
    border-top: 2px solid #ce2828;
    border-right: 2px solid #ce2828;
    content: '';
    display: block;
    float: right;
    margin-top: 15px;
    margin-right: 2px;
    width: 5px;
    height: 5px;

    transform: ${(props: Object) => renderArrowIconRotate(props)};
  }
`);
