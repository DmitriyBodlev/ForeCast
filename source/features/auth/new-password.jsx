import React from 'react';
import { connect } from 'react-redux';
import { compose, withHandlers, pure } from 'recompose';
import { createStructuredSelector } from 'reselect';
import { reduxForm, Field } from 'redux-form';
import { makeSelectLocale } from '../locales/selectors';
import { makeSelectAuthData } from '../auth/selectors';
import { required } from '../../utils/validator';
import { renderLogoLink } from '../../utils/global-helpers';
import { renderInputGroup, AlertDanger } from '../../ui';
import { sendCreateNewPasswordRequest } from './actions';
import {
  FormWrapper,
  FormContainer,
  FormLogo,
  StyledForm,
  StyledFormTitle,
  StyledButton } from './ui';

const qs = require('qs');

export const enhance = compose(
  reduxForm({
    form: 'newPasswordForm',
  }),
  withHandlers({
    handleSubmitNew: (props: Object) => (values: any) => {
      const params = qs.parse(props.location.search, { ignoreQueryPrefix: true });
      props.sendCreateNewPasswordRequest({ ...values, ...params });
    },
  }),
  pure,
);

export const NewPassword = (props: Object) => (
  <FormWrapper >
    <FormContainer>
      <FormLogo>
        xcvxcbcxbxcb
      </FormLogo>
      <StyledForm onSubmit={props.handleSubmit(props.handleSubmitNew)}>
        <StyledFormTitle>{props.locale['features:auth:titles:new-password']}</StyledFormTitle>
        <Field
          errorFontSize={10}
          errorMargin='0 0 5px 0'
          width={1}
          margin='10px 0px 10px 0px'
          padding='0px 26px'
          fontSize={14}
          name='oldPassword'
          component={renderInputGroup}
          type='password'
          placeholder={props.locale['features:auth:password-old']}
          validate={required} />
        <Field
          errorFontSize={10}
          errorMargin='0 0 5px 0'
          width={1}
          margin='10px 0px 10px 0px'
          padding='0px 26px'
          fontSize={14}
          name='newPassword'
          component={renderInputGroup}
          type='password'
          placeholder={props.locale['features:auth:password-new']}
          validate={required} />
        <StyledButton
          width={1}
          my='10'
          p='8px 16px'
          fontSize={16}
          type='submit'
          disabled={props.pristine || !props.valid || props.submitting}
        >
          {props.locale['features:auth:btn:send']}
        </StyledButton>
      </StyledForm>
      { props.auth.createNewPasswordError &&
        <AlertDanger>{props.auth.createNewPasswordError}</AlertDanger> }
    </FormContainer >
  </FormWrapper>
);

const mapStateToProps = (state: Object) => (createStructuredSelector({
  locale: makeSelectLocale(state),
  auth: makeSelectAuthData(state),
}));

export default connect(mapStateToProps, {
  sendCreateNewPasswordRequest,
})(enhance(NewPassword));
