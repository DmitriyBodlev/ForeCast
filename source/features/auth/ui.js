import styled from 'react-emotion';
import { Form } from 'redux-form';
import { Button, LinkTo } from '../../ui';

const renderBackgroundLink = (props: Object) => {
  if (props.currentTheme === 'dark') {
    return 'https://s3.amazonaws.com/amous-front-dev/images/dark-background.svg';
  }
  return 'https://s3.amazonaws.com/amous-front-dev/images/light-background.svg';
};

export const FormWrapper = styled.div`
  align-items: center;
  background-image: url(${(props: Object) => renderBackgroundLink(props)});
  background-repeat: no-repeat;
  background-attachment: fixed;
  background-position: center; 
  background-size: cover;
  display: flex;
  overflow: auto;
  height: 100vh;
  width: 100vw;
`;

export const FormContainer = styled.div`
  margin-left: 25%;
  width: 50%;

  & input {
    height: 42px;

    &[type='text']:focus, &[type='password']:focus, &[type='email']:focus {
      box-shadow: ${(props: Object) => props.theme.auth.forms.inputs.boxShadow};
    }
  }

  & .remember label {
    font-weight: 500;
    margin-left: 5px;
    color: ${(props: Object) => props.theme.auth.forms.labels.color};
  }
`;

export const FormLogo = styled.div`
  text-align: center;
  margin-bottom: 20px;

  & img {
    max-width: 100%;
  }
`;

export const StyledForm = styled(Form)`
  padding: 30px 48px;
  background-color: ${(props: Object) => props.theme.auth.forms.bgColor};
`;

export const StyledFormTitle = styled.div`
  font-size: 18px;
  font-weight: bold;
  margin-bottom: 5px;;

  color: ${(props: Object) => props.theme.auth.forms.titleColor};
`;

export const StyledFormSubtitle = styled.div`
  font-size: 14px;
  margin-bottom: 21px;

  color: ${(props: Object) => props.theme.auth.forms.titleColor};
`;

export const StyledButton = styled(Button)`
  font-weight: bold;
  height: 42px;

  background-color: ${(props: Object) => props.theme.auth.forms.buttons.bgColor};
  color: ${(props: Object) => props.theme.auth.forms.buttons.textColor};
`;

export const StyledLinkTo = styled(LinkTo)`
  font-weight: 500;
  margin-right: 15px;

  color: ${(props: Object) => props.theme.auth.forms.labels.color};
`;

