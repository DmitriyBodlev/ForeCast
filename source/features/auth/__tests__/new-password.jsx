import React from 'react';
import { shallow } from 'enzyme';
import { NewPassword } from '../new-password';
import { Form } from '../../../ui';
import locale from '../../../locale-default';

describe('features/auth/new-password', () => {
  const handleSubmit = jest.fn();

  it('should render new password form', () => {
    const wrapper = shallow(
      <NewPassword
        handleSubmit={handleSubmit}
        auth={{createNewPasswordError: null}}
        locale={locale} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render AlertDanger component', () => {
    const wrapper = shallow(
      <NewPassword
        handleSubmit={handleSubmit}
        auth={{createNewPasswordError: 'error'}}
        locale={locale} />);

    const formComponent = wrapper.find(Form);

    expect(wrapper).toMatchSnapshot();
    expect(formComponent).toBeTruthy();
  });

  it('should handle submit form', () => {
    const handleSubmit = jest.fn();
    const wrapper = shallow(
      <NewPassword
        handleSubmit={handleSubmit}
        auth={{createNewPasswordError: null}}
        locale={locale} />);

    wrapper.find(Form).simulate('submit');

    expect(handleSubmit).toBeCalled();

    expect(wrapper).toMatchSnapshot();
  });
});
