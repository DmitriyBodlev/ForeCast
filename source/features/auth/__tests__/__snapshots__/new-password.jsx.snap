// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features/auth/new-password should handle submit form 1`] = `
<styled(Component)>
  <styled(Component)>
    <styled(Component)>
      <img
        alt="Logo"
        src="https://static.wixstatic.com/media/c55a42_78b53de3385f4c32baacc030ddaad4b5~mv2_d_2000_1500_s_2.png/v1/fill/w_361,h_196,al_c,usm_0.66_1.00_0.01/c55a42_78b53de3385f4c32baacc030ddaad4b5~mv2_d_2000_1500_s_2.png"
      />
    </styled(Component)>
    <styled(Component)>
      <Field
        component={[Function]}
        errorFontSize={14}
        errorMargin="0 0 5px 0"
        label="Old password:"
        labelFontSize={14}
        labelMargin="5px 0 5px 0"
        name="oldPassword"
        type="password"
        validate={[Function]}
        width={1}
      />
      <Field
        component={[Function]}
        errorFontSize={14}
        errorMargin="0 0 5px 0"
        label="New password:"
        labelFontSize={14}
        labelMargin="5px 0 5px 0"
        name="newPassword"
        type="password"
        validate={[Function]}
        width={1}
      />
      <WithTheme(styled(Component))
        disabled={true}
        fontSize={16}
        my="10"
        p="8px 16px"
        type="submit"
      >
        Send
      </WithTheme(styled(Component))>
      <styled(Link)
        to="login"
      >
        Login
      </styled(Link)>
    </styled(Component)>
  </styled(Component)>
</styled(Component)>
`;

exports[`features/auth/new-password should render AlertDanger component 1`] = `
<styled(Component)>
  <WithTheme(styled(Component))>
    <styled(Component)>
      <img
        alt="Logo"
        src="https://s3.amazonaws.com/amous-front-dev/logo/undefined-logo.svg"
      />
    </styled(Component)>
    <WithTheme(styled(Form))>
      <WithTheme(styled(Component)) />
      <Field
        component={[Function]}
        errorFontSize={10}
        errorMargin="0 0 5px 0"
        fontSize={14}
        margin="10px 0px 10px 0px"
        name="oldPassword"
        padding="0px 26px"
        placeholder="Old password"
        type="password"
        validate={[Function]}
        width={1}
      />
      <Field
        component={[Function]}
        errorFontSize={10}
        errorMargin="0 0 5px 0"
        fontSize={14}
        margin="10px 0px 10px 0px"
        name="newPassword"
        padding="0px 26px"
        placeholder="New password"
        type="password"
        validate={[Function]}
        width={1}
      />
      <WithTheme(styled(styled(Component)))
        disabled={true}
        fontSize={16}
        my="10"
        p="8px 16px"
        type="submit"
        width={1}
      >
        SEND NEW PASSWORD
      </WithTheme(styled(styled(Component)))>
    </WithTheme(styled(Form))>
    <styled(styled(Component))>
      error
    </styled(styled(Component))>
  </WithTheme(styled(Component))>
</styled(Component)>
`;

exports[`features/auth/new-password should render new password form 1`] = `
<styled(Component)>
  <WithTheme(styled(Component))>
    <styled(Component)>
      <img
        alt="Logo"
        src="https://s3.amazonaws.com/amous-front-dev/logo/undefined-logo.svg"
      />
    </styled(Component)>
    <WithTheme(styled(Form))>
      <WithTheme(styled(Component)) />
      <Field
        component={[Function]}
        errorFontSize={10}
        errorMargin="0 0 5px 0"
        fontSize={14}
        margin="10px 0px 10px 0px"
        name="oldPassword"
        padding="0px 26px"
        placeholder="Old password"
        type="password"
        validate={[Function]}
        width={1}
      />
      <Field
        component={[Function]}
        errorFontSize={10}
        errorMargin="0 0 5px 0"
        fontSize={14}
        margin="10px 0px 10px 0px"
        name="newPassword"
        padding="0px 26px"
        placeholder="New password"
        type="password"
        validate={[Function]}
        width={1}
      />
      <WithTheme(styled(styled(Component)))
        disabled={true}
        fontSize={16}
        my="10"
        p="8px 16px"
        type="submit"
        width={1}
      >
        SEND NEW PASSWORD
      </WithTheme(styled(styled(Component)))>
    </WithTheme(styled(Form))>
  </WithTheme(styled(Component))>
</styled(Component)>
`;
